File: app/src/main/java/me/gabeg/sicksends/sport/SsSportRepository.java
package me.gabeg.sicksends.sport;

import android.app.Application;

import me.gabeg.sicksends.problem.SsProblemDatabase;
import me.gabeg.sicksends.problem.SsProblemRepository;

/**
 * Repository for sport problems.
 */
public class SsSportRepository
	extends SsProblemRepository<SsSportDao, SsSport>
{

	/**
	 */
	public SsSportRepository(Application app)
	{
		super(app);
	}

	/**
	 * @see SsProblemRepository#buildDao(SsProblemDatabase)
	 */
	@Override
	protected void buildDao(SsProblemDatabase db)
	{
		this.mProblemDao = db.sportDao();
	}

}

File: app/src/main/java/me/gabeg/sicksends/sport/SsSportRepository.kt
package me.gabeg.sicksends.sport

import me.gabeg.sicksends.problem.SsGenericProblemRepository
import javax.inject.Inject

/**
 * Repository for sport problems.
 */
class SsSportRepository @Inject constructor(dao : SsSportDao)
	: SsGenericProblemRepository<SsSportProblem>(dao)
